astropy/time/core.py

```diff
@@ -437,6 +438,9 @@ def precision(self, val):
             raise ValueError('precision attribute must be an int between '
                              '0 and 9')
         self._precision = val
+        if hasattr(self, '_time'):
```

Comment:
I thought the way the code worked it should not be necessary ever to set the `_time` internals. But I now see that `precision` is not propagated if one uses `.value`
```
In [14]: t = Time('2015-12-30')

In [15]: t.value
Out[15]: '2015-12-30 00:00:00.000'

In [16]: t.precision = 9

In [17]: t.value
Out[17]: '2015-12-30 00:00:00.000'
```
I think the "bug" is in `value`, though, not in setting these properties... Note that it doesn't show up in `repr`:
```
In [19]: t
Out[19]: <Time object: scale='utc' format='iso' value=2015-12-30 00:00:00.000000000>
```
because that shows `getattr(self, self.format)`. This suggests that one way out would be to redefine
```
    @property
    def value(self):
        return getattr(self, self.format)
```
and then put the existing definition in `__getattr__` (obviously, `__repr__` can then use `self.value`).